cogify/handler.py

```diff
@@ -64,13 +64,13 @@ def upload_file(outfilename, collection):
         raise
 
 
-def download_file(file_uri: str):
+def download_file(file_uri: str, earthdata_username=None, earthdata_password=None):
     filename = os.path.splitext(os.path.basename(file_uri))[0]
     filename = f"/tmp/{filename}"
     if "http" in file_uri:
         # This isn't working for GPMIMERG, need to use .netrc
```

Comment:
@amarouane-ABDELHAK this could be because auth headers are not forwarded when Earthdata redirects your download requests. Here is a ticket on that: https://stackoverflow.com/a/58126432/512111

Reply:
I changed the function signature because it will be used for other AWS services (Airflow, lambda, ECS...) and we need it not to depend on the environment variables.
In this case the environment variables will override the function parameter.